modules/nf-core/graphtyper/genotype/main.nf

```diff
@@ -0,0 +1,45 @@
+process GRAPHTYPER_GENOTYPE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::graphtyper=2.7.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/graphtyper:2.7.2--h7d7f7ad_0':
+        'quay.io/biocontainers/graphtyper:2.7.2--h7d7f7ad_0' }"
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+    path ref
+    path ref_fai
+    val region
```

Comment:
The `region` has to be supplied via `task.ext.args` as it's not a mandatory input value. Is there an option where the tool can take a BED file or interval file? If so you can change the region input to this?

Reply:
It looks like it is optional from the help info, but it seems to be required. When trying to run the program without it I get the following error:

```
[2023-01-19 10:16:17.449] <error> No regions specified. Either use --region or --region_file option to specify regions.
```

It does have an option for storing the region in a file with the format `${chr}:${begin}-${end}`, one per line. I can change the input format to that. The documentation does not mention compatibility with BED files.

Reply:
Yeah that's fine, maybe add that file as an optional input and mention in the `meta.yml` that either this file or `--region` has to be supplied

Reply:
I dont see how to add an optional input. It seems that modules can only have positional parameters currently? Do you mean rely on users to pass one of the two using `task.ext.args`? Since they both do the same thing, I could also just use `--region_file` as the only way to specify the region, since `--region` does the same thing except only allows for a single input. `--region` is just a way to make it easier to look at a single region when using the tool interactively on the command line, so I am not sure it is very useful in the context of a module.  It seems like having a single way to specify the region, either as a file or a list of values (which the module would convert to a file), would be less confusing and less likely to result in user error.

Reply:
Yeah my bad for not being clear but this was what I meant. You can add a check in the module itself that checks if a regions file is given or if `--region` is in the `task.ext.args`. If neither is the case you can make it throw an error.

Reply:
Ok, that makes sense. This should be done now.